server/coordinator/scheduler/rebalanced_shard_scheduler.go

```diff
@@ -48,12 +52,22 @@ func (r RebalancedShardScheduler) Schedule(ctx context.Context, clusterSnapshot
 			if err != nil {
 				return ScheduleResult{}, err
 			}
-			return ScheduleResult{
-				Procedure: p,
-				Reason:    fmt.Sprintf("the shard:%d on the node:%s does not meet the balance requirements,it should be assigned to node:%s", shardNode.ID, shardNode.NodeName, node.Node.Name),
-			}, nil
+			procedures = append(procedures, p)
+			reasons.WriteString(fmt.Sprintf("the shard:%d on the node:%s does not meet the balance requirements,it should be assigned to node:%s. ", shardNode.ID, shardNode.NodeName, node.Node.Name))
```

Comment:
Put the parameters at the end, then you can grep can grep the msg when troubleshooting.